src/parser/mod.rs

```diff
@@ -4139,6 +4160,194 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse statements of the DropTrigger type such as:
+    ///
+    /// ```sql
+    /// DROP TRIGGER [ IF EXISTS ] name ON table_name [ CASCADE | RESTRICT ]
+    /// ```
+    pub fn parse_drop_trigger(&mut self) -> Result<Statement, ParserError> {
+        if !dialect_of!(self is PostgreSqlDialect | GenericDialect) {
+            self.prev_token();
+            return self.expected("an object type after DROP", self.peek_token());
+        }
+        let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
+        let trigger_name = self.parse_object_name(false)?;
+        self.expect_keyword(Keyword::ON)?;
+        let table_name = self.parse_object_name(false)?;
+        let option = self
+            .parse_one_of_keywords(&[Keyword::CASCADE, Keyword::RESTRICT])
+            .map(|keyword| match keyword {
+                Keyword::CASCADE => ReferentialAction::Cascade,
+                Keyword::RESTRICT => ReferentialAction::Restrict,
+                _ => unreachable!(),
+            });
+        Ok(Statement::DropTrigger {
+            if_exists,
+            trigger_name,
+            table_name,
+            option,
+        })
+    }
+
+    pub fn parse_create_trigger(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+        if !dialect_of!(self is PostgreSqlDialect | GenericDialect) {
+            self.prev_token();
+            return self.expected("an object type after CREATE", self.peek_token());
+        }
+
+        let name = self.parse_object_name(false)?;
+        let period = self.parse_trigger_period()?;
+
+        let events = self.parse_keyword_separated(Keyword::OR, Parser::parse_trigger_event)?;
+        self.expect_keyword(Keyword::ON)?;
+        let table_name = self.parse_object_name(false)?;
+
+        // [ NOT DEFERRABLE | [ DEFERRABLE ] [ INITIALLY IMMEDIATE | INITIALLY DEFERRED ] ]
+        let mut deferrable: Option<bool> = None;
+
+        if self.parse_keyword(Keyword::NOT) {
+            self.expect_keyword(Keyword::DEFERRABLE)?;
+            deferrable = Some(false);
+        } else if self.parse_keyword(Keyword::DEFERRABLE) {
+            deferrable = Some(true);
+        };
+
+        let initially: Option<DeferrableInitial> = (deferrable.is_some()
+            && self.parse_keyword(Keyword::INITIALLY))
+        .then(|| {
+            Ok::<_, ParserError>(
+                match self.expect_one_of_keywords(&[Keyword::IMMEDIATE, Keyword::DEFERRED])? {
+                    Keyword::IMMEDIATE => DeferrableInitial::Immediate,
+                    Keyword::DEFERRED => DeferrableInitial::Deferred,
+                    _ => unreachable!(),
+                },
+            )
+        })
+        .transpose()?;
```

Comment:
we can also flatten this? a regular if else would suffice i imagine

Reply:
Done